04-onion-routing.md

```diff
@@ -367,6 +368,102 @@ otherwise meets the amount criterion (eg. some other failure, or
 invoice timeout), however if it were to fulfill only some of them,
 intermediary nodes could simply claim the remaining ones.
 
+### Onion Messages
+
+Onion messages have an onion with an alternate `hop_payload`
```

Comment:
Please drop the reference to `hop_payload` and just specify the whole message. It takes a ton of reading to maybe kinda understand how the message is laid out on the wire, instead of being able to look at something.

Reply:
Having recently scrolled around up and down to figure out the message format, strongly agree here.

Reply:
Yeah, this gets simpler when we remove the legacy format altogether.  But I've rewritten it to duplicate the full description of the onion (with references back, since you don't need to reimplement it but should share code!).